### ISSUE TITLE
After delete SimObject, the memory is still not free all.

### ISSUE THREAD

**Issue by kynora** [GarageGames/Torque3D#1789]
I try to run simulate create 100000 SimObjects. Each object have fields value.
All object consume about 40 mb of memory, after delete all objects. only 30mb are free and still remaining about 10mb.
Are those fields values still keep in memory?

This bugs effect on my  game server performance, the memory usage increase day by day until I restart game server.

```javascript
new SimGroup(NewPointerGroup);
function SimulateNewPointer(%max)
{
   if (%max $= "")
      NewPointerGroup.deleteAllObjects();
   else
   {
      for (%i =0; %i < %max; %i++)
      {
         %obj = new SimObject()
         {
            value1 = 123456;
            value2 = 678901;
            value3 = "sjldfljkskjldfjksdfkjlskfljsdkjlfsdkjlfsdfj slfjslkdfs";
         };
         NewPointerGroup.add(%obj);
      }
   }
}
```

**Comment by Areloch** [comment]
Thanks for spotting this. If we've got a memory leak somewhere, we'll definitely want to plug that!

**Comment by crabmusket** [comment]
If the strings are different on each object, they'll hang around in the StringTable, but that shouldn't account for _that_ much memory.

**Comment by Areloch** [comment]
Did some initial testing on this and can affirm it's not a real memory leak, all this looks to be getting cleaned up when the program closes. Which means that it's just the engine holding onto stuff and not letting go when an object is deleted.

I'll keep poking at this to see if I can narrow the problem spot further.

**Comment by kynora** [comment]
Can we replace using StringTable with mIdString to normal string? Each time SimObject register it create char buffer and store in StringTable.

char idBuffer[64];
        dSprintf(idBuffer, sizeof(idBuffer), "%d", mId);
        mIdString = StringTable->insert( idBuffer );

**Comment by Areloch** [comment]
I'm not sure how much that'd impact anything though. Did you test it on your end and see an improvement?

**Comment by kynora** [comment]
Yes, mIdString and mInternalName of SimObject cause increase memroy overtime. When we set object internal name or find object by internal name the unique names are store in global StringTable.
